dev_guide/builds/build_environment.adoc

```diff
@@ -0,0 +1,84 @@
+[[dev-guide-build-build-environment]]
+= Build Environment
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+
+[[overview]]
+== Overview
+
+[NOTE]
+====
+You can also manage environment variables defined in the `BuildConfig` with the
+xref:../../dev_guide/environment_variables.adoc#dev-guide-environment-variables[`oc set env`] command.
+====
+
+[[using-valuefrom-references]]
+== Using valueFrom References
+
+[[using-secrets-as-environment-variables]]
+=== Using Secrets as Environment Variables
+
+[source,yaml]
+----
+env:
+  - name: SECRETKEYREF_ENV
+    valueFrom: <1>
+      secretKeyRef: <2>
+        name: mysecret <3>
+        key: username <4>
+----
+<1> the value should be resolved using the following references
+<2> the following `name` and `key` refer to a `secret`
+<3> the name of the `secret`
+<4> the name of the `key` in the `secret`
+
+[[using-configmaps-as-environment-variables]]
+=== Using ConfigMaps as Environment Variables
+
+[source,yaml]
+----
+env:
+  - name: CONFIGMAPKEYREF_ENV
+    valueFrom: <1>
+      configMapKeyRef: <2>
+        name: myconfigmap <3>
+        key: mykey <4>
+----
+
+<1> the value should be resolved using the following references
+<2> the following `name` and `key` refer to a `configMap`
+<3> the name of the `configMap`
+<4> the name of the `key` in the `configmap`
+
+[[using-build-fields-as-environment-variables]]
+=== Using Build Fields as Environment Variables
+
+[source,yaml]
+----
+env:
+  - name: FIELDREF_ENV
+    valueFrom: <1>
+      fieldRef: <2>
+        fieldPath: metadata.name <3>
+----
+
+<1> the value should be resolved using the following references
+<2> the following `fieldPath` refers to a jsonpath in the `build`
+<3> the jsonpath to the field in the `build`
+
+[[using-container-resources-as-environment-variables]]
+=== Using Container Resources as Environment Variables
+
+Referencing container resources using valueFrom in build environment variables is currently not supported
+as the references are resolved before the container is created.
+
+
```

Comment:
I had previously removed it via your comment: https://github.com/openshift/openshift-docs/pull/4557#discussion_r121723411